core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -345,7 +346,7 @@ class SparkConf(loadDefaults: Boolean)
 
   /** Set a name for your application. Shown in the Spark web UI. */
   def setAppName(name: String): SparkConf = {
-    set("spark.app.name", name)
+    set("spark.app.name", StringEscapeUtils.escapeHtml4(name))
```

Comment:
I'm not sure this is a safe change. Since `spark.app.name` is used in many places inside and outside of Apache Spark, this kind of implicit change may cause significant breaking changes due to the mismatches.

Let me ask in this way: if we throws `Exception` here, do you think it's okay?

Reply:
Agree with @dongjoon-hyun.
Instead, it would be much better to simply escape this in Spark UI, so that what gets returned works for html.